### PR TITLE
Add dashboard table view

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Adds dummy views for all asset types (project, directory, secret, file)
  - The backend metadata does not exist for many of these columns.

### Important Notes
To do:
- [x] figure out the best way to add the custom classes - tailwind likes to avoid custom classes whenever possible

### Screenshots
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by somebody1234** [comment]
other notes:
- nctdt's `projectManagerAdapter` is a good idea, so consider adding it here?
- `dashboard.tsx` is *very* long, so it may be a good idea to split it.
  - however, it's not necessarily a good idea to split it into multiple components, especially if they will only be used in `dashboard.tsx`

**Comment by somebody1234** [comment]
not quite sure what the "9 new commits" thing is all about. the IDE component isn't here (as expected) though so everything should still be fine

**Comment by somebody1234** [comment]
or not... i merged the other PR into this one, will be fixing and force-pushing

**Comment by PabloBuchu** [comment]
@somebody1234 please update with the changes from newest develop. I can't build ide on macos

**Review by PabloBuchu — CHANGES_REQUESTED** [review]
@somebody1234 some adjustments:
- change all font sizes and weights to correspond with the figma
- the same with distances between `Drive` label and directory tree indicators and others
- can you add fixed paths and shared with so we can see how the whole path controller looks like?

**Comment by somebody1234** [comment]
- distances: not changed yet
- font sizes: changed
- font weights: should already be correct
- fixed paths: directory support is already there - i forgot that directories can't be added through UI yet, will add next commit
- fixed shared: added

**Comment by somebody1234** [comment]
all adjustments should be made now @PabloBuchu

**Comment by PabloBuchu** [comment]
@somebody1234 are we able to open this for review?

**Comment by somebody1234** [comment]
@PabloBuchu this PR technically depends on #6005, however that one is *very* small so i think it might be a good idea to mark this one as ready as well.

**Comment by somebody1234** [comment]
#6005 should be merged into this to clean up the diff - i'll do it after this `./run backend ci-check` is done (ci-check is failing on #6212 for linux but not other OSes, for some reason)

**Comment by wdanilo** [comment]
Let's keep it as draft until its ready to merge pls :)

**Comment by PabloBuchu** [comment]
@somebody1234 I think this one we can open for review, right?

**Comment by PabloBuchu** [comment]
also @somebody1234  one thing from now on we should be able to see new dashboard in electron rather than standard IDE view. Can you add a flag (like `-authentication`, e.x. `-cloud-dashboard`) so we can run Enso with cloud based dashboard?

basically we don't want to hide `#dashboard` and also we won't we running `void appInstance.run()` either

**Comment by PabloBuchu** [comment]
@somebody1234 one thing to fix. After clean `./run ide build` I am getting errors while connecting to project manager (see screenshot below). Interesting thing is that it only occur on **first** run after building the package

To reproduce:
1. Comment out `lib/content/src/index:ts:148` and `lib/content/src/index:ts:169` 
2. Build enso with: `./run ide build --skip-version-check --skip-wasm-opt --wasm-profile=dev --backend-source release --backend-release nightly`
3. Run with: `./dist/ide/mac-arm64/Enso.app/Contents/MacOS/Enso -authentication=true -debug.dev-tools`

Screenshot 2023-04-11 at 11 01 25

**Comment by somebody1234** [comment]
- yes, will open soon after i merge
- i was assuming `-authentication` is supposed to be `-cloud-dashboard`
  - either way i was looking into it because i need a way to test changes to IDE styles so that the IDE fits inside the dashboard instead of e.g. covering it. it seems like pretty extensive changes will be required so i'm thinking of maybe delaying it for another PR, if not a completely separate PR
- re: the error, i will investigate soon

**Comment by somebody1234** [comment]
oh actually i'll need to add a screenshot to this before i reopen

**Comment by PabloBuchu** [comment]
> * i was assuming `-authentication` is supposed to be `-cloud-dashboard`

At the beginning yes, but I think now we can add more granularity since the libs will start using authentication heavily but not necessarily the new dashboard (yet..)

maybe it should be a group called `cloud` with `authentication` and `dashboard`

this way you can use `-cloud.authentication=true` and `-cloud.dashboard=true`

**Comment by somebody1234** [comment]
and i should also probably change `#dashboard` to `.dashboard`, since the high specificity `#dashboard` makes tailwind preflight override `react-hot-toast`'s styles

**Comment by somebody1234** [comment]
still to do:
- add new option group
- investigate bug
- open PR

**Comment by somebody1234** [comment]
for posterity (since new commits have changed line numbers):
`lib/content/src/index.ts:148` is
https://github.com/enso-org/enso/blob/3de3aabd10585a20d8d4b5ca5aac275468631920/app/ide-desktop/lib/content/src/index.ts#L148

`lib/content/src/index.ts:169` is the last line here:
https://github.com/enso-org/enso/blob/3de3aabd10585a20d8d4b5ca5aac275468631920/app/ide-desktop/lib/content/src/index.ts#L167-L169

File: app/ide-desktop/lib/content-config/src/config.json [hunk 1]
```diff
@@ -150,4 +158,4 @@
       }
     }
   }
-}
+}
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/content-config/src/config.json]: no nl

File: app/ide-desktop/lib/content/src/index.html [hunk 2]
```diff
@@ -40,7 +40,7 @@
     <script type="module" src="/run.js" defer></script>
   </head>
   <body>
-    <div id="dashboard"></div>
+    <div id="dashboard" class="dashboard"></div>
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/content/src/index.html]: can we change id and class to `enso-dashboard`? then we probably need also update prefix in tailwindcss

File: app/ide-desktop/lib/dashboard/src/index.tsx [hunk 3]
```diff
@@ -13,13 +13,9 @@ const ESBUILD_PATH = '/esbuild'
 const ESBUILD_EVENT_NAME = 'change'
 
 // ===================
-// === Live reload ===
+// === Live relaod ===
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/index.tsx]: `reload`

**Comment by PabloBuchu** [comment]
Thanks for the flags. Please address comments and open for review

**Comment by somebody1234** [comment]
@PabloBuchu can't repro, sorry - logs show that the project manager is being started:
```
│ Loading the window address 'http://localhost:8080?startup.platform=linux&cloud.authentication=true&engine.preferredVersion=2023.1.1-nightly.2023.4.11'.
╰ Done in 2481.7 ms.
[info] [2023-04-12T04:46:01.6Z] [org.enso.projectmanager.boot.ProjectManager$] Starting Project Manager...
[info] [2023-04-12T04:46:02.032Z] [org.enso.projectmanager.boot.ProjectManager$] Started server at 127.0.0.1:30535, press enter to kill server
```
(did a fresh rebuild just to double-check)
that said, this is using the new flags instead of commenting likes out, that might affect things

**Comment by PabloBuchu** [comment]
@somebody1234 so one minor issue: `-cloud.authentication=true` doesn't show login templates without `-cloud.dashboard=true` 

listing projects from project_manager works fine, but clicking on open project calls for API. We could leave it for now and fix it in https://github.com/enso-org/enso/pull/5804 this PR

**Comment by somebody1234** [comment]
@PabloBuchu login templates?
if you mean the dashboard UI, that's intentional - otherwise i'm not quite sure what having a separate `dashboard` option is supposed to do
(right now, there is code to explicitly only show the dashboard if `cloud.dashboard=true`

**Comment by PabloBuchu** [comment]
> @PabloBuchu login templates? if you mean the dashboard UI, that's intentional - otherwise i'm not quite sure what having a separate `dashboard` option is supposed to do (right now, there is code to explicitly only show the dashboard if `cloud.dashboard=true`

@somebody1234 
so when we use authentication the user should be welcomed with login page (sign in with github etc) but dashboard flag should indicate whether he lands on old IDE (appInstance.run in onAuthenticated method) or see the new dashboard with new table view introduced by this PR

**Comment by somebody1234** [comment]
@PabloBuchu a bit late but i pushed a commit that should fix it
(for posterity, it's because `DASHBOARD_PATH` is `/`, i.e. the dashboard is the default route - and it navigates away to the authentication route(s) when the user is not logged in. one of the previous commits disabled this route entirely; this one adds it back but makes it blank)

**Comment by PabloBuchu** [comment]
@somebody1234 did you pushed your latest changes? Because I still can't get to work cloud flags properly
1. `-cloud.authentication=true` should allow user for signing in and then redirect to old IDE. Instead I am getting a blank page

https://user-images.githubusercontent.com/2855109/231709224-832d42d9-2623-4c38-9fb7-f424a8071bcd.mov

2. If I was logged in previously the flag works fine and I see the old IDE

Uploading Screen Recording 2023-04-13 at 11.00.53.mov…


3. `-cloud.authentication=true -cloud.dashboard=true` looks like it works fine

https://user-images.githubusercontent.com/2855109/231709985-e8719d83-ec12-4fcd-ba53-ec3bbd12a8b7.mov

**Comment by somebody1234** [comment]
whoops :|
pushed now @PabloBuchu

**Comment by PabloBuchu** [comment]
Thanks works now.

QA:
- flags: 🟢 
- build & run: 🟢 
- ide watch: 🟢 
- gui watch: 🟢 

@wdanilo can you please take a look and review the code?

**Review by PabloBuchu — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx [hunk 4]
```diff
@@ -74,14 +76,26 @@ export const SET_USERNAME_PATH = '/set-username'
...
 // ===========
 
-/** Global configuration for the `App` component. */
-export interface AppProps {
-    /** Logger to use for logging. */
+interface BaseAppProps {
     logger: loggerProvider.Logger
     platform: platformModule.Platform
+    /** Whether the dashboard should be rendered. */
+    enableDashboard: boolean
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx]: minor thing, but I think `showDashboard` would be a better name. `enableDashboard` makes me confused what "enabling" it really means, while, according to docs, this is only about showing it.

**Review by wdanilo — APPROVED** [review]

**Pull request merged**